11219 adapt UI font size

harry924 opened the pull request:
closes https://github.com/JabRef/jabref/issues/11219
Merged all the CSS files to Base.css  

converted all px units to em  

Fixed toggle button and sperator padding issue  

Conducted UI testing by comparing interfaces before and after changes.   





### Mandatory checks

<!-- 
- Go through the list below. Please don't remove any items.
- [x] done; [ ] not done / not applicable
-->

- [x] I own the copyright of the code submitted and I licence it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.


![Separator-issue](https://github.com/user-attachments/assets/d4db30c5-bf47-4de3-9e86-4b8dfe51c12d)

![Fixed-Separator](https://github.com/user-attachments/assets/0de41ac6-9236-499f-94bd-c1eb8ce461f3)

![misaligned-filter-icon](https://github.com/user-attachments/assets/94f6a06f-dafc-47a2-bfc2-0b4a2b9775e1)

![Fixed-filter-icon](https://github.com/user-attachments/assets/9c27ed1d-c00a-401b-85b1-92380f012609)

Siedlerchr commented:
then remove the unused css files as well

Siedlerchr reviewed (changes requested):
delete the unused css files and check if they are referenced somehwere

calixtus commented on src/main/java/org/jabref/gui/BaseBackUp.css:
```diff
@@ -0,0 +1,1736 @@
+.root {
```
You really don't have to create backup files, this is what git is for.

harry924 commented:
Hi,  I have deleted original individual CSS files and the BaseBackUp.css and consolidated styles into Base.css.
and also updated all the relevant Java classes to reference Base.css instead of the deleted files.

koppor commented on docs/_config.yml:
```diff
@@ -3,7 +3,7 @@ title: "Developer Documentation"
 color_scheme: light
 
-exclude: [_config.yml, .dockerignore, .gitignore, CNAME, Dockerfile, Gemfile, Gemfile.lock, README.md, mkdocs-custom.css]
+exclude: [_config.yml, .dockerignore, .gitignore, CNAME, Dockerfile, Gemfile, Gemfile.lock, README.md]
```
Even if this affects our documentation, I think, this removal is OK.

koppor replied:
Keep this.

Reason:

```yaml
site_name: JabRef Developer Documentation
repo_url: https://github.com/jabref/jabref/
theme:
    name: readthedocs
extra_css:
    - mkdocs-custom.css
```

Siedlerchr commented on src/main/java/org/jabref/gui/groups/GroupTreeView.java:
```diff
@@ -117,7 +117,7 @@ public GroupTreeView(TaskExecutor taskExecutor,
 
         createNodes();
-        this.getStylesheets().add(Objects.requireNonNull(GroupTreeView.class.getResource("GroupTree.css")).toExternalForm());
+        this.getStylesheets().add(Objects.requireNonNull(GroupTreeView.class.getResource("/org/jabref/gui/Base.css")).toExternalForm());
```
I don't think this is even necessary, the Base.cs stylesheet is applied globally, so it would be automatically applied to all nodes in the hierarchy. However, you probably need to add custom classes for the group tree etc to prevent side effects with other elements 

see my PR for an example 
https://github.com/JabRef/jabref/pull/11881

koppor replied:
i think  @harry924 this means, remove this line.

koppor commented on src/main/java/org/jabref/gui/mergeentries/newmergedialog/ThreeWayMergeView.java:
```diff
@@ -47,7 +47,7 @@ public class ThreeWayMergeView extends VBox {
         this.preferences = preferences;
 
-        getStylesheets().add(ThreeWayMergeView.class.getResource("ThreeWayMergeView.css").toExternalForm());
+        getStylesheets().add(ThreeWayMergeView.class.getResource("/org/jabref/gui/Base.css").toExternalForm());
```
Remove this line.

(And check if JabRef still looks good)

calixtus replied:
@HoussemNasri can you take a look?

koppor commented on docs/mkdocs-custom.css:
```diff
@@ -1,7 +0,0 @@
-li.toctree-l2:first-child {
```
keep this file

HoussemNasri commented:
The changes look good to me. However, the pull request does more than just adapt font size. For instance, the merging of all CSS styles into `Base.css` could be a source of regressions, so I vouch for adding multiple entries to the changelog or merge the individual commits without squash.

Siedlerchr commented:
Overall looks good, however, JabRef opens always maximized, even if I resize it and move it to the screen center but this might be another issue because I reset my prefs as well

Siedlerchr commented:
Changelog should contain a note about the effect of custom themes!

calixtus commented:
@InAnYan I took the liberty to simplify the privacy component and to make it a bit more coherent to the mrdlib privacy dialog. Can you also take a look?

InAnYan reviewed (dismissed):
![image](https://github.com/user-attachments/assets/a1d6a01c-bb48-4432-892d-e4ed87c1e4c7)

Good! There are some small issues

calixtus commented:
![grafik](https://github.com/user-attachments/assets/5fe99f68-378b-4024-bdc6-af45a3b40262)

calixtus commented:
GPT4all privacy link was the same as before. Change it in another pr as this is focused on css and ui
